module/main.c

```diff
@@ -781,10 +752,12 @@ static void handler_HidTimer(s32 data) {
                             r_edit_dirty |= R_ALL;
                         }
                         else if (mode == M_TRACK) {
-                            if (tele_patterns[edit_pattern]
-                                    .v[edit_index + offset_index] < 32766) {
-                                tele_patterns[edit_pattern]
-                                    .v[edit_index + offset_index]++;
+                            int16_t v = tele_get_pattern_val(
+                                edit_pattern, edit_index + offset_index);
+                            if (v < 32766) {
+                                tele_set_pattern_val(edit_pattern,
+                                                     edit_index + offset_index,
+                                                     v++);
```

Comment:
something like `tele_constrain_pattern_val` could be a good function to have as it'll make it more readable and will keep the magic number in the place where all pattern related things are. i can add it once this is merged.